dev-libs/nspr/patches/nspr-4.26.patchset

```diff
@@ -0,0 +1,324 @@
+ #elif defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) \
+-    || defined(FREEBSD)
++    || defined(FREEBSD) || defined(__HAIKU__)
+ #define PT_PRIO_MIN            sched_get_priority_min(SCHED_OTHER)
+ #define PT_PRIO_MAX            sched_get_priority_max(SCHED_OTHER)
+ #elif defined(NTO)
+@@ -177,7 +177,7 @@ extern int (*_PT_aix_yield_fcn)();
+     || defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) \
+     || defined(FREEBSD) || defined(NETBSD) || defined(OPENBSD) \
+     || defined(BSDI) || defined(NTO) || defined(DARWIN) \
+-    || defined(UNIXWARE) || defined(RISCOS)
++    || defined(UNIXWARE) || defined(RISCOS) || defined(__HAIKU__)
+ #define _PT_PTHREAD_YIELD()             sched_yield()
+ #else
+ #error "Need to define _PT_PTHREAD_YIELD for this platform"
+diff --git a/pr/include/md/_unixos.h b/pr/include/md/_unixos.h
+index 7d444cd..0cc560b 100644
+--- a/pr/include/md/_unixos.h
++++ b/pr/include/md/_unixos.h
+@@ -15,7 +15,7 @@
+  * not be redefined.
+  */
+ #if !defined(LINUX) && !defined(__GNU__) && !defined(__GLIBC__) \
+-    && !defined(DARWIN)
++    && !defined(DARWIN) && !defined(__HAIKU__)
+ #ifndef FD_SETSIZE
+ #define FD_SETSIZE  4096
+ #endif
+@@ -46,7 +46,7 @@
+  */
+ #include <sys/time.h>
+ #include <sys/types.h>
+-#if defined(AIX)
++#if defined(AIX) || defined(__HAIKU__)
+ #include <sys/select.h>
+ #endif
+ 
+diff --git a/pr/include/md/prosdep.h b/pr/include/md/prosdep.h
+index 42eba3d..e372315 100644
+--- a/pr/include/md/prosdep.h
++++ b/pr/include/md/prosdep.h
+@@ -49,7 +49,7 @@ PR_BEGIN_EXTERN_C
+ #elif defined(HPUX)
+ #include "md/_hpux.h"
+ 
+-#elif defined(LINUX) || defined(__GNU__) || defined(__GLIBC__)
++#elif defined(LINUX) || defined(__GNU__) || defined(__GLIBC__) || defined(__HAIKU__)
+ #include "md/_linux.h"
+ 
+ #elif defined(DARWIN)
+diff --git a/pr/include/obsolete/protypes.h b/pr/include/obsolete/protypes.h
+index 5400afb..65eb0f5 100644
+--- a/pr/include/obsolete/protypes.h
++++ b/pr/include/obsolete/protypes.h
+@@ -9,6 +9,9 @@
+  * possible time. The NSPR API is implemented and documented using
+  * the new definitions.
+  */
++#if defined(__HAIKU__)
++#include <SupportDefs.h>
++#endif
+ 
+ #if !defined(PROTYPES_H)
+ #define PROTYPES_H
+@@ -35,7 +38,7 @@ typedef PRIntn intn;
+  *
+  * On OS/2, sys/types.h defines uint.
+  */
+-#if defined(XP_UNIX) || defined(XP_OS2)
++#if defined(XP_UNIX) || defined(XP_OS2) || defined(__HAIKU__)
+ #include <sys/types.h>
+ #endif
+ 
+@@ -48,7 +51,7 @@ typedef PRIntn intn;
+  * uint
+  */
+ 
+-#if !defined(XP_OS2) && !defined(XP_UNIX) || defined(NTO)
++#if !defined(XP_OS2) && !defined(XP_UNIX) && !defined(__HAIKU__) || defined(NTO)
+ typedef PRUintn uint;
+ #endif
+ 
+@@ -56,13 +59,15 @@ typedef PRUintn uint;
+  * uint64
+  */
+ 
++#if !defined(__HAIKU__)
+ typedef PRUint64 uint64;
++#endif
+ 
+ /*
+  * uint32
+  */
+ 
+-#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO)
++#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO) && defined(__HAIKU__)
```

Comment:
This is so far as I found the only changed line, but this should be the same as it is used on the other patchset

Reply:
This should be changed to reflect the one from the _x86 patchset:
``diff
-+#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO) && defined(__HAIKU__)
++#if !defined(_WIN32) && !defined(XP_OS2) && !defined(NTO) && !defined(__HAIKU__)
```